ddtrace/contrib/sqlalchemy/engine.py

```diff
@@ -47,33 +63,61 @@ def __init__(self, tracer, service, engine):
             app=self.vendor,
             app_type=sqlx.APP_TYPE)
 
+        # attach the PIN
+        Pin(
+            app=self.vendor,
+            tracer=tracer,
+            service=self.service,
+            app_type=sqlx.APP_TYPE,
+        ).onto(engine)
+
         listen(engine, 'before_cursor_execute', self._before_cur_exec)
         listen(engine, 'after_cursor_execute', self._after_cur_exec)
         listen(engine, 'dbapi_error', self._dbapi_error)
 
     def _before_cur_exec(self, conn, cursor, statement, *args):
-        span = self.tracer.trace(
+        pin = Pin.get_from(self.engine)
+        if not pin or not pin.enabled():
+            # don't trace the execution
+            return
+
+        span = pin.tracer.trace(
             self.name,
-            service=self.service,
+            service=pin.service,
             span_type=sqlx.TYPE,
-            resource=statement)
+            resource=statement,
+        )
 
         if not _set_tags_from_url(span, conn.engine.url):
             _set_tags_from_cursor(span, self.vendor, cursor)
 
     def _after_cur_exec(self, conn, cursor, statement, *args):
-        span = self.tracer.current_span()
+        pin = Pin.get_from(self.engine)
+        if not pin or not pin.enabled():
+            # don't trace the execution
+            return
+
+        span = pin.tracer.current_span()
         if not span:
             return
 
         try:
             if cursor and cursor.rowcount >= 0:
                 span.set_tag(sqlx.ROWS, cursor.rowcount)
+            elif cursor and cursor.rowcount == -1:
```

Comment:
this check has been added otherwise a test fails; according to previous tests, we expect to have the `sql.rows` field set even for 0 rows. In the case of SQLite, this patch is required.